src/mango_cursor.erl

```diff
@@ -30,29 +31,20 @@
 
 create(Db, Selector0, Opts) ->
     Selector = mango_selector:normalize(Selector0),
-
-    ExistingIndexes = mango_idx:list(Db),
-    if ExistingIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, no_indexes_defined})
-    end,
-
-    FilteredIndexes = maybe_filter_indexes(ExistingIndexes, Opts),
-    if FilteredIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, no_index_matching_name})
-    end,
-
-    SortIndexes = mango_idx:for_sort(FilteredIndexes, Opts),
-    if SortIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, missing_sort_index})
-    end,
-
-    UsableFilter = fun(I) -> mango_idx:is_usable(I, Selector) end,
-    UsableIndexes = lists:filter(UsableFilter, SortIndexes),
-    if UsableIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, selector_unsupported})
-    end,
-
-    create_cursor(Db, UsableIndexes, Selector, Opts).
+    UsableIndexes = mango_idx:get_usable_indexes(Db, Selector0, Opts),
+
+    {use_index, IndexSpecified} = proplists:lookup(use_index, Opts),
+    case {length(UsableIndexes), length(IndexSpecified)} of
+        {0, 1} ->
+            ?MANGO_ERROR({no_usable_index, selector_unsupported});
+        {0, 0} ->
+            AllDocs = mango_idx:special(Db),
+            io:format("Index ~p~n", [AllDocs]),
+            create_cursor(Db, AllDocs, Selector, Opts);
```

Comment:
@tonysun83 I tried to follow your suggestion to pass the AllDocs index to avoid adding an additional selector. Given I obtain the All Docs index (the special index) and pass it my app crashes as `columns` for the special index is hardcoded to `_id` and no other fields are allowed: https://github.com/apache/couchdb-mango/blob/master/src/mango_idx_special.erl#L62-L63 which leaves my without a defined range in https://github.com/apache/couchdb-mango/blob/master/src/mango_cursor_view.erl#L33

```
2016-01-27 15:26:09.833 [error] node1@127.0.0.1 <0.3711.0> req_err(3969470405) unknown_error : function_clause
    [<<"mango_idx_special:start_key/1 L72">>,<<"mango_cursor_view:execute/3 L83">>,<<"mango_httpd:handle_find_req/2 L185">>,<<"mango_httpd:handle_req/2 L37">>,<<"chttpd:process_request/1 L290">>,<<"chttpd:handle_request_int/1 L227">>,<<"mochiweb_http:headers/6 L122">>,<<"proc_lib:init_p_do_apply/3 L237">>]
```